pkg/gadgets/utils/test/runner.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+// Config defines how the runner should behave.
+// TODO: We could implement more options like unsharing the network
+// namespace, running on a different group ID, etc.
+type Config struct {
+	// User ID to run under
+	UID int
+}
+
+type Info struct {
+	Pid       int
+	Comm      string
+	UID       int
+	MountNsID uint64
+}
+
+// Runner is a helper type to execute tests in different conditions. It
+// creates a go routine that is executed in a different mount namespace,
+// user ID, etc. to simulate events hapenning inside containers.
+type Runner struct {
+	config *Config
+
+	tasks   chan func() error
+	replies chan error
+	Info    *Info
+}
+
+func NewRunner(config *Config) (*Runner, error) {
+	if config == nil {
+		config = &Config{}
+	}
+	r := &Runner{
+		config:  config,
+		tasks:   make(chan func() error),
+		replies: make(chan error),
+	}
+
+	go r.run()
```

Comment:
Is it possible to force a go routine to use a new thread? If not, then the isolation of calls won't be guaranteed.

Reply:
Ah, this is cool, TIL!